lib/mods/date.rb

```diff
@@ -0,0 +1,390 @@
+require 'edtf'
+
+module Mods
+  class Date
+    attr_reader :xml
+
+    ##
+    # Ugly date factory that tries to pick an appropriate parser for the
+    # type of data.
+    #
+    # @param [Nokogiri::XML::Element] xml A date-flavored MODS field from the XML
+    # @return [Mods::Date]
+    def self.from_element(xml)
+      case xml.attr(:encoding)
+      when 'w3cdtf'
+        Mods::Date::W3cdtfFormat.new(xml)
+      when 'iso8601'
+        Mods::Date::Iso8601Format.new(xml)
+      when 'marc'
+        Mods::Date::MarcFormat.new(xml)
+      when 'edtf'
+        Mods::Date::EdtfFormat.new(xml)
+      # when 'temper'
```

Comment:
Not supported? Should this be removed?

Reply:
Stanford doesn't have any examples of this, but it's called out as one of the supported MODS formats. I figured a comment was the best way to leave a clue about where one might add support if they cared. 🤷‍♂️